api/envoy/extensions/filters/network/ext_proc/v3/ext_proc.proto

```diff
@@ -60,6 +61,9 @@ message NetworkExternalProcessor {
     lte {seconds: 3600}
     gte {}
   }];
+
+  // Options related to the sending and receiving of dynamic metadata.
+  MetadataOptions metadata_options = 5;
```

Comment:
There will be a merge conflict with this PR #39265 whichever is merged at first.